waddrmgr/manager.go

```diff
@@ -1360,15 +1363,31 @@ func (m *Manager) Unlock(passphrase []byte) error {
 	return nil
 }
 
-// MarkUsed updates the used flag for the provided address id.
-func (m *Manager) MarkUsed(addressID []byte) error {
+// fetchUsed returns true if the provided address id was flagged used.
+func (m *Manager) fetchUsed(addressID []byte) (bool, error) {
+	var used bool
+	err := m.namespace.View(func(tx walletdb.Tx) error {
+		used = fetchAddressUsed(tx, addressID)
+		return nil
+	})
+	return used, err
+}
+
+// MarkUsed updates the used flag for the provided address.
+func (m *Manager) MarkUsed(address btcutil.Address) error {
+	addressID := address.ScriptAddress()
```

Comment:
@jrick I think this was the external API call you mentioned which takes `addressID`